pkg/chains/formats/intotoite6/intotoite6.go

```diff
@@ -58,20 +59,59 @@ func (i *InTotoIte6) Wrap() bool {
 }
 
 func (i *InTotoIte6) CreatePayload(obj interface{}) (interface{}, error) {
-	var tr *v1beta1.TaskRun
 	switch v := obj.(type) {
 	case *v1beta1.TaskRun:
-		tr = v
+		return i.generateAttestationFromTaskRun(v)
+	case *v1beta1.PipelineRun:
+		return i.generateAttestationFromPipelinerun(v)
 	default:
 		return nil, fmt.Errorf("intoto does not support type: %s", v)
 	}
-	return i.generateAttestationFromTaskRun(tr)
+}
+
+// There most likely exists a better place to put this, adding as another method for now
+// Possibly a generic attestation interface?
+func (i *InTotoIte6) generateAttestationFromPipelinerun(pr *v1beta1.PipelineRun) (interface{}, error) {
+	// We don't need to pass in a client/context here, as we only want access to the original object
+	// Need a better way to differentiate between an abstracted original k8s object and getting the latest values
+	// - Possibly pass client and context in each method call instead of adding it to the struct?
+	pro := objects.NewPipelineRunObject(pr, nil, nil)
+
+	subjects := GetSubjectDigests(pro, i.logger)
+
+	// Adding a sample value to know the attestation is an example
+	invocation := slsa.ProvenanceInvocation{}
+	params := make(map[string]string)
+	params["message"] = "This is an example attestation for pipelineruns, we'll fill in the details later"
+	invocation.Parameters = params
+
+	att := intoto.ProvenanceStatement{
+		StatementHeader: intoto.StatementHeader{
+			Type:          intoto.StatementInTotoV01,
+			PredicateType: slsa.PredicateSLSAProvenance,
+			Subject:       subjects,
+		},
+		Predicate: slsa.ProvenancePredicate{
+			Builder: slsa.ProvenanceBuilder{
+				ID: i.builderID,
```

Comment:
Hmmm so this is interesting, because IIUC the builder id is supposed to be an identifier for "the thing that ran the build" - but this is actually ambiguous because a Pipeline by itself doesn't actually run anything - it farms things out to Tasks to execute. 🤔 

Maybe this should be some reference to the controller that orchestrated the run, but that's also ambiguous if multiple controllers happen to pick up the same pipeline (i.e. retries, etc.). 🤔 🤔 🤔 

Don't need to solve this in this PR - but if you have any thoughts I'd love to discuss more.